### PR TITLE
Upgrade to Jetty 11

### PR DESCRIPTION
**Pull request by adamretter**
Requires Java 17 support to be merged first - https://github.com/eXist-db/exist/pull/4574  (and so at this time it contains all commits of that PR also).

* This required me to also release a new version of Apache XML-RPC which supports Jakarta 5 - https://search.maven.org/artifact/com.evolvedbinary.thirdparty.org.apache.xmlrpc/xmlrpc/5.0.0/pom
* This also required me to release an updated version of Milton WebDav which supports Jakarta 5 - https://search.maven.org/artifact/org.exist-db.thirdparty.com.ettrema/milton/1.8.1.3-jakarta5/pom

### CODE REVIEW

File: exist-jetty-config/src/main/resources/org/exist/jetty/etc/webapps/portal/WEB-INF/web.xml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 <web-app 
    xmlns="http://xmlns.jcp.org/xml/ns/javaee" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/web-app_4_0.xsd"
    metadata-complete="false"
-   version="3.1">
+   version="5.0">
```
Comment (reinhapa) [inline exist-jetty-config/src/main/resources/org/exist/jetty/etc/webapps/portal/WEB-INF/web.xml:7]: I think this should refer to the Jakarta schemas from now on. See https://jakarta.ee/xml/ns/jakartaee/ for more details on this.

File: exist-jetty-config/src/main/resources/org/exist/jetty/etc/webdefault.xml [hunk 2]
```diff
@@ -2,9 +2,9 @@
 <web-app 
    xmlns="http://xmlns.jcp.org/xml/ns/javaee" 
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/web-app_4_0.xsd"
    metadata-complete="false"
-   version="3.1"> 
+   version="5.0">
```
Comment (reinhapa) [inline exist-jetty-config/src/main/resources/org/exist/jetty/etc/webdefault.xml:7]: I think this should refer to the Jakarta schemas from now on. See https://jakarta.ee/xml/ns/jakartaee/ for more details on this.

File: exist-jetty-config/src/main/resources/standalone-webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -9,9 +9,9 @@
 <web-app 
     xmlns="http://xmlns.jcp.org/xml/ns/javaee" 
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/web-app_4_0.xsd"
     metadata-complete="false"
-    version="3.1">
+    version="5.0">
```
Comment (reinhapa) [inline exist-jetty-config/src/main/resources/standalone-webapp/WEB-INF/web.xml:14]: I think this should refer to the Jakarta schemas from now on. See https://jakarta.ee/xml/ns/jakartaee/ for more details on this.

File: exist-jetty-config/src/main/resources/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -8,9 +8,9 @@
 <web-app 
     xmlns="http://xmlns.jcp.org/xml/ns/javaee" 
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/web-app_4_0.xsd"
     metadata-complete="false"
-    version="3.1">
+    version="5.0">
```
Comment (reinhapa) [inline exist-jetty-config/src/main/resources/webapp/WEB-INF/web.xml:13]: I think this should refer to the Jakarta schemas from now on. See https://jakarta.ee/xml/ns/jakartaee/ for more details on this.

**Review by reinhapa — CHANGES_REQUESTED** [review]
Why are those casts from `ByteBuffer` to `Buffer` as `ByteBuffer` already extends from it? I'm a bit confused here. This would reduce the number of changed files significantly though:

```
exist-core/src/main/java/org/exist/storage/blob/BlobStoreDumpTool.java
exist-core/src/main/java/org/exist/storage/blob/BlobStoreImpl.java
exist-core/src/main/java/org/exist/storage/journal/Journal.java
exist-core/src/main/java/org/exist/storage/journal/JournalReader.java
exist-core/src/main/java/org/exist/storage/lock/FileLock.java
exist-core/src/main/java/org/exist/storage/statistics/DataGuide.java
exist-core/src/main/java/org/exist/util/io/ByteBufferInputStream.java
exist-core/src/main/java/org/exist/util/io/MemoryMappedFileFilterInputStreamCache.java
exist-core/src/main/java/org/exist/xquery/value/BinaryValueFromFile.java
exist-core/src/test/java/org/exist/storage/journal/JournalTest.java
exist-core/src/test/java/org/exist/storage/ValueIndexFactoryTest.java
```

**Comment by adamretter** [comment]
@reinhapa A lot of your comments are duplicated from https://github.com/eXist-db/exist/pull/4574 as this PR is stacked atop that one as it is required.

File: .idea/runConfigurations/Java_Admin_Client.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="Java Admin Client" type="Application" factoryName="Application">
-    <option name="ALTERNATIVE_JRE_PATH" value="1.8" />
+    <option name="ALTERNATIVE_JRE_PATH" value="liberica-17" />
```
[the remark this answers is not in the document]
Comment (adamretter) [reply .idea/runConfigurations/Java_Admin_Client.xml:3]: I have pushed an additional commit which addresses this.

File: .idea/runConfigurations/Jetty_Server.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="Jetty Server" type="Application" factoryName="Application">
-    <option name="ALTERNATIVE_JRE_PATH" value="1.8" />
+    <option name="ALTERNATIVE_JRE_PATH" value="liberica-17" />
```
[the remark this answers is not in the document]
Comment (adamretter) [reply .idea/runConfigurations/Jetty_Server.xml:3]: I have pushed an additional commit which addresses this.

File: .idea/runConfigurations/Local_Jetty_Server.xml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="Local Jetty Server" type="Application" factoryName="Application">
-    <option name="ALTERNATIVE_JRE_PATH" value="1.8" />
+    <option name="ALTERNATIVE_JRE_PATH" value="liberica-17" />
```
[the remark this answers is not in the document]
Comment (adamretter) [reply .idea/runConfigurations/Local_Jetty_Server.xml:3]: I have pushed an additional commit which addresses this.

File: exist-docker/src/main/resources-filtered/README.md [hunk 8]
```diff
@@ -17,7 +17,7 @@ The images are based on Google Cloud Platform's ["Distroless" Docker Images](htt
 *   [Docker](https://www.docker.com): `18-stable`
 ### For building
 *   [maven](https://maven.apache.org/): `^3.6.0`
-*   [java](https://www.java.com/): `8`
+*   [java](https://www.java.com/): `17`
```
[the remark this answers is not in the document]
Comment (adamretter) [reply exist-docker/src/main/resources-filtered/README.md:20]: I think the choice of distro is a personal choice. I don't myself use Temurin, but instead use BellSoft's Liberica, or something Azul's Zulu.

If you feel strongly that this link should be changed, I will not object, but could it be done in a separate PR from you?

File: exist-jetty-config/src/main/resources/webapp/WEB-INF/web.xml (hunk 4 above)
Comment (adamretter) [reply exist-jetty-config/src/main/resources/webapp/WEB-INF/web.xml:13]: @reinhapa Good shout! I have pushed an additional commit that addresses this.

File: exist-jetty-config/src/main/resources/standalone-webapp/WEB-INF/web.xml (hunk 3 above)
Comment (adamretter) [reply exist-jetty-config/src/main/resources/standalone-webapp/WEB-INF/web.xml:14]: @reinhapa Good shout! I have pushed an additional commit that addresses this.

File: exist-jetty-config/src/main/resources/org/exist/jetty/etc/webdefault.xml (hunk 2 above)
Comment (adamretter) [reply exist-jetty-config/src/main/resources/org/exist/jetty/etc/webdefault.xml:7]: @reinhapa Good shout! I have pushed an additional commit that addresses this.

File: exist-jetty-config/src/main/resources/org/exist/jetty/etc/webapps/portal/WEB-INF/web.xml (hunk 1 above)
Comment (adamretter) [reply exist-jetty-config/src/main/resources/org/exist/jetty/etc/webapps/portal/WEB-INF/web.xml:7]: @reinhapa Good shout! I have pushed an additional commit that addresses this.

File: extensions/security/iprange/src/main/java/org/exist/security/realm/iprange/IPRangeServlet.java [hunk 9]
```diff
@@ -24,18 +24,17 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.exist.http.servlets.ExistExtensionServlet;
-import org.exist.security.AbstractRealm;
 import org.exist.security.AuthenticationException;
 import org.exist.security.SecurityManager;
 import org.exist.security.Subject;
 import org.exist.xquery.XQueryContext;
 
-import javax.servlet.ServletConfig;
```
Comment (dizzzz) [inline extensions/security/iprange/src/main/java/org/exist/security/realm/iprange/IPRangeServlet.java:33]: from my memories this servlet could and should be removed; the original requester does not use it anymore....

**Review by dizzzz — APPROVED** [review]
LGTM, one servlet could be removed (checking now with PieterL.)

**Review by reinhapa — APPROVED** [review]

**Comment by dizzzz** [comment]
image

**Comment by adamretter** [comment]
@dizzzz Where are you seeing that error?

**Comment by dizzzz** [comment]
> @dizzzz Where are you seeing that error?

https://github.com/eXist-db/exist/actions/runs/3154141844/jobs/5131407589

At least the Ubuntu build, error last lines

**Comment by adamretter** [comment]
> https://github.com/eXist-db/exist/actions/runs/3154141844/jobs/5131407589
> At least the Ubuntu build, error last lines

@dizzzz I think that must be an old run, the current one is passing just fine.

**Comment by line-o** [comment]
As we learned from #4586 uploading smaller payloads encoded as `multipart/form-data` is not tested. This test should be added as part of this change to ensure this is still working.

**Pull request merged**